src/aleph/vm/orchestrator/views/__init__.py

```diff
@@ -251,8 +251,9 @@ async def update_allocations(request: web.Request):
 
     # First free resources from persistent programs and instances that are not scheduled anymore.
     allocations = allocation.persistent_vms | allocation.instances
-    for execution in pool.get_persistent_executions():
-        if execution.vm_hash not in allocations:
+    # Make a copy since the pool is modified
+    for execution in list(pool.executions.values()):
+        if execution.vm_hash not in allocations and execution.is_running:
```

Comment:
Comparing with `get_persistent_executions()`:

```py
    def get_persistent_executions(self) -> Iterable[VmExecution]:
        for _vm_hash, execution in self.executions.items():
            if execution.persistent and execution.is_running:
                yield execution
```
why not just wrap it in a `list` before iterating?

Reply:
Yeah  that works too, good suggestion

Reply:
Please add it to this PR ;-)

Reply:
Who is this addressed to?
This PR is already merged so we will need to make another one